weaviate/collections/classes/batch.py

```diff
@@ -165,26 +175,56 @@ class BatchObjectReturn:
             A boolean indicating whether or not any of the objects in the batch failed to be inserted. If this is `True`, then the `errors` dictionary will contain at least one entry.
     """
 
-    all_responses: List[Union[uuid_package.UUID, ErrorObject]]
+    _all_responses: List[Union[uuid_package.UUID, ErrorObject]]
     elapsed_seconds: float
+    """The time taken to perform the batch operation."""
     errors: Dict[int, ErrorObject]
+    """A dictionary of all the failed responses from the batch operation. The keys are the indices of the objects in the overall batch, and the values are the `Error` objects."""
     uuids: Dict[int, uuid_package.UUID]
+    """A dictionary of all the successful responses from the batch operation. The keys are the indices of the objects in the overall batch, and the values are the `uuid_package.UUID` objects."""
     has_errors: bool = False
+    """A boolean indicating whether or not any of the objects in the batch failed to be inserted. If this is `True`, then the `errors` dictionary will contain at least one entry."""
+
+    @property
+    def all_responses(self) -> List[Union[uuid_package.UUID, ErrorObject]]:
+        """@deprecated: A list of all the responses from the batch operation. Each response is either a `uuid_package.UUID` object or an `Error` object.
+
+        WARNING: Now that only the last `MAX_STORED_RESULTS` uuids are stored in the `uuids` dictionary, the keys of the `errors` and `uuids` dictionaries can no longer be used to map the indices in the `all_responses` field!
+        """
+        _Warnings.batch_results_objects_all_responses_attribute()
+        return self._all_responses
 
     def __add__(self, other: "BatchObjectReturn") -> "BatchObjectReturn":
-        self.all_responses += other.all_responses
+        self._all_responses += other._all_responses
 
         prev_max = max(self.errors.keys()) if len(self.errors) > 0 else -1
         for k1, v1 in other.errors.items():
-            self.errors[prev_max + k1] = v1
+            self.errors[prev_max + k1 + 1] = v1
```

Comment:
It was a problem of defining the original index. Since `prev_max` starts at `-1` and `k1` also starts at `0` then the first entry to `.errors` will be `-1`, will add one though!